cityofaustin/vision-zero
# Add summary statistics to location details and rework column metadata types

## Associated issues

* https://github.com/cityofaustin/atd-data-tech/issues/20354
* https://github.com/cityofaustin/atd-data-tech/issues/20316

<img width="1214" alt="Screenshot 2024-12-19 at 1 10 51 PM" src="https://github.com/user-attachments/assets/dc009f0c-a216-493d-9af5-d46fa2a12a18" />

This PR restructures the typing and config structure for managing editable columns in the app. It makes our config + components more abstract, so that they can better handle arbitrary relationships rather than the hardcoded "lookup table" relationship.

One additional change that I would like to make, but have not in the interest of the code diff, is to stop using the word "column" where we really mean "field". This came up in some PR feedback from Chia and Rose recently. 

Thank you for reviewing! Since this is so large, I'd be happy to set up a time to pair or review as a group interactively.

## Testing

**URL to test:** [Netlify](https://deploy-preview-1634--vision-zero-nextjs.netlify.app/crashes)

**Steps to test:**

1. Navigate to a location's details page and verify that the **Details** card is rendering normally.
2. Head over to a crash details page, and try editing a variety of fields, including fields that use a relationship (e.g. **Collision Type**), a yes/no field, and fields in the **Units** table.



---

#### Ship list
